src/core/stream.js

```diff
@@ -1073,21 +1076,12 @@ var Jbig2Stream = (function Jbig2StreamClosure() {
     var jbig2Image = new Jbig2Image();
 
     var chunks = [];
-    var decodeParams = this.dict.getArray('DecodeParms', 'DP');
-
-    // According to the PDF specification, DecodeParms can be either
-    // a dictionary, or an array whose elements are dictionaries.
-    if (isArray(decodeParams)) {
-      if (decodeParams.length > 1) {
-        warn('JBIG2 - \'DecodeParms\' array with multiple elements ' +
```

Comment:
That is covered by `Note also that we currently do not handle the case where DecodeParms is an Array entirely correct in the various image Streams, and this patch fixes that for free.` from the commit message.
The code in question was written by me, in PR #4003, back when I didn't know as much about the PDF format as I do now. As explained in the [PDF specification](http://www.adobe.com/content/dam/Adobe/en/devnet/acrobat/pdfs/PDF32000_2008.pdf#G6.1919337) there should be *one* `DecodeParms` entry per `Filter` (if applicable), hence the current code will *not work correctly* if the `JBIG2Decode` entry isn't the *first* of a list of `Filter`s. This is now instead handled for us by the pre-existing code in [`Parser_filter`](https://github.com/mozilla/pdf.js/blob/f209030b9967fdcd335369ce53954df803f021c5/src/core/parser.js#L546-L563).

With regards to https://github.com/mozilla/pdf.js/pull/7727/files#diff-349869351c851c049016fb108cf3104bR543; that was just me adding a tiny bit of code in an attempt to avoid a possible future bug (given all the weird things PDF generators tend to do). 
Please note that it's not necessary for any of our existing test-cases to pass, but I figured that it cannot hurt!
This concerns the (perhaps theoretical) case where the `Filter` entry contains a standard `Name`, but the `DecodeParms` entry contains `[ {some Dict} ]` (i.e. a `Dict` inside of an Array) instead of just normal `Dict` as it should be. Again I'm not sure if that can happen.

I hope this addresses your questions! If so, would you be comfortable with reviewing this PR?